JLio.Commands/Logic/AddPutLogic.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.Linq;
+using JLio.Core;
+using JLio.Core.Contracts;
+using JLio.Core.Extensions;
+using JLio.Core.Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace JLio.Commands.Logic
+{
+    public abstract class AddPut : CommandBase
```

Comment:
I can't agree with the current version of this class.
Relying on "is Add" to decide to do Add or Put,  and also having the concrete implementation as private methods for both of these operations breaks a lot of clean principles (single responsibility and open-closed).

My suggestion is to introduce abstract methods for handling objects and handling arrays,  and le the "Add" and the "Put" provide the implementations for them.
That will make the code much clearer. 

I also suggest renaming this from "AddPut" to something less concrete, as there might be other implementations in the future that  can provide implementation of the newly introduced abstract methods, which might not be necessarily Add or Put, but something with a different name that follows a similar pattern of doing something different for Objects and Arrays

Reply:
Thanks for the comments. I fixed the code style and added the set command into the refactoring because it is pretty similar. If you can, have a look again. ;-)